packages/runtime-core/src/apiCreateApp.ts

```diff
@@ -397,7 +400,7 @@ export function createAppAPI<HostElement>(
       runWithContext(fn) {
         currentApp = app
         try {
-          return fn()
+          return scope.run(fn) as ReturnType<typeof fn>
```

Comment:
Isn't this enough?

```suggestion
          return scope.run(fn)!
```

We could technically also run the function raw if the scope isn't running but I don't think anybody should run `app.runWithContext()` if the app is unmounted

Reply:
Given the nature of the api (advanced), I don't think it's that helpful, it could end up adding noise to end users and being confusing.